client/src/js/wall/FirstUser.js

```diff
@@ -1,67 +1,78 @@
-import { noop } from "lodash-es";
+import { get } from "lodash-es";
 import React from "react";
 import { connect } from "react-redux";
-import { BoxGroupHeader, BoxGroupSection, Button, Input, InputGroup, InputLabel, PasswordInput } from "../base";
+import { Formik, Form, Field } from "formik";
+import {
+    BoxGroupHeader,
+    BoxGroupSection,
+    Button,
+    Input,
+    InputGroup,
+    InputLabel,
+    InputError,
+    PasswordInput
+} from "../base";
 import { createFirstUser } from "../users/actions";
 import { WallContainer, WallDialog, WallDialogFooter } from "./Container";
 
-export class FirstUser extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            username: "",
-            password: ""
-        };
-    }
-
-    handleChange = e => {
-        const { name, value } = e.target;
-        this.setState({
-            [name]: value
-        });
+export const FirstUser = ({ onSubmit, generalError, usernameErrors, passwordErrors }) => {
+    const initialValues = {
+        username: "",
+        password: ""
     };
```

Comment:
Should be defined at module-level. As it is, this will be redefined every render.